src/test/java/uk/co/ramp/people/PersonTest.java

```diff
@@ -0,0 +1,266 @@
+        person.updateStatus(EXPOSED, timeNow, exposer);
+
+        Assert.assertEquals(EXPOSED, person.getStatus());
+        Assert.assertTrue(person.getNextStatusChange() > timeNow);
+
+    }
+
+    @Test(expected = InvalidStatusTransitionException.class)
+    public void checkTime() throws NoSuchFieldException {
+
+        int time = rnd.nextInt(0, 100);
+        int flatTransition = (int) diseaseProperties.getMeanTimeToInfected();
+
+        person.updateStatus(EXPOSED, time - flatTransition);
+
+        person.checkTime(time);
+
+        flatTransition = (int) diseaseProperties.getMeanTimeToRecovered();
+
+        Assert.assertEquals(INFECTED, person.getStatus());
+        Assert.assertEquals(time + flatTransition, person.getNextStatusChange());
+
+
+        time += flatTransition;
+        person.checkTime(time);
+        Assert.assertEquals(RECOVERED, person.getStatus());
+        Assert.assertEquals(-1, person.getNextStatusChange());
+
+        person.updateStatus(SUSCEPTIBLE, ++time);
+        FieldSetter.setField(person, person.getClass().getDeclaredField("nextStatusChange"), time);
+
+        try {
+            person.checkTime(time);
+            Assert.fail();
+        } catch (InvalidStatusTransitionException e) {
+            Assert.assertThat(appender.getOutput(), containsString(String.format("Changing status from RECOVERED for person.id %d is not a valid transition", id)));
+            throw e;
+        }
+    }
+
+    @Test
+    public void notNow() {
+        int time = rnd.nextInt(0, 100);
+        int flatTransition = (int) diseaseProperties.getMeanTimeToInfected();
+
+        person.updateStatus(EXPOSED, time - flatTransition);
+        person.checkTime(time - 10);
+
+        // event times don't match, so nothing happens
+        Assert.assertEquals("", appender.getOutput());
+    }
+
+    @Test
+    public void randomExposure() {
+        int time = rnd.nextInt(0, 1000);
+        person.randomExposure(time);
+        Assert.assertThat(appender.getOutput(), containsString(String.format("Person with id: %d has been randomly exposed at time %d", id, time)));
+    }
+
+
+    @Test(expected = InvalidStatusTransitionException.class)
+    public void invalidExposure() {
+        int time = rnd.nextInt(0, 1000);
+
+        System.out.println(person.getStatus());
+        person.updateStatus(EXPOSED, 0);
+        time = person.getNextStatusChange();
+
+        person.checkTime(time);
+        System.out.println(person.getStatus());
+
+        try {
+            person.randomExposure(time);
+        } catch (InvalidStatusTransitionException e) {
+            String message = String.format("The person with id: %d should not be able to transition from %s to %s", person.getId(), person.getStatus(), EXPOSED);
+            Assert.assertThat(appender.getOutput(), containsString(message));
+            throw e;
+        }
+
+    }
+
+    @Test
+    public void getGaussianDistributionValue() {
+        int n = 10000;
```

Comment:
I don't understand this test (but I haven't spent long trying to figure it out).

Reply:
(and similarly for the following couple of tests)

Reply:
They are just checking the rng is getting something around the mean value for a large number of runs. Just to make sure the distributions behave.